runtime/bcutil/ROMClassBuilder.cpp

```diff
@@ -457,13 +485,18 @@ ROMClassBuilder::prepareAndLaydown( BufferManager *bufferManager, ClassFileParse
 		return classFileOracle.getBuildResult();
 	}
 
-	SRPKeyProducer srpKeyProducer(&classFileOracle);
```

Comment:
ifdef

Reply:
Should I create the ifdef for the SRPKeyProducer like the following:
`
#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
	SRPKeyProducer srpKeyProducer(&classFileOracle, &_interfaceInjectionInfo);
#else
	SRPKeyProducer srpKeyProducer(&classFileOracle);
#endif /* J9VM_OPT_VALHALLA_VALUE_TYPES */
`
and similarly do ifdef and else for the constructor?

Reply:
Yes, like that.

the else should also of the comment:

```
#else /* J9VM_OPT_VALHALLA_VALUE_TYPES */
```

Reply:
Do I want to be doing this with RomClassWriter and the Helper private class?

Reply:
Yes, for anything where you had to change the APIs